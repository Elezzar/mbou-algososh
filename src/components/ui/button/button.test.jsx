import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import {Button} from './button';

describe('Button', () => {
  it('Кнопка с текстом отрисовывается корректно', () => {
    const tree = renderer.create(<Button text='test' />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Кнопка без текста отрисовывается корректно', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Заблокированная кнопка отрисовывается корректно', () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Кнопка с индикатором загрузки отрисовывается корректно', () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Колбэк при клике на кнопку вызывается корректно', () => {
    const mockFn = jest.fn();
    render(<Button onClick={mockFn} />)
    fireEvent.click(screen.getByRole('button'));
    expect(mockFn).toHaveBeenCalled();
  })
})