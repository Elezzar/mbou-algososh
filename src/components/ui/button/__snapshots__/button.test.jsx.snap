// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Заблокированная кнопка отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Кнопка без текста отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Кнопка с индикатором загрузки отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Кнопка с текстом отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;
